experimental/packages/sdk-logs/src/LogRecord.ts

```diff
@@ -73,15 +75,26 @@ export class LogRecord implements ReadableLogRecord {
 
   constructor(logger: Logger, logRecord: logsAPI.LogRecord) {
     const {
-      timestamp = Date.now(),
+      timestamp,
+      observedTimestamp,
       severityNumber,
       severityText,
       body,
       attributes = {},
       context,
     } = logRecord;
 
-    this.hrTime = timeInputToHrTime(timestamp);
+    if (timestamp && observedTimestamp) {
+      this.hrTime = timeInputToHrTime(timestamp);
+      this.hrTimeObserved = timeInputToHrTime(observedTimestamp);
+    } else {
+      const now = hrTime();
+      this.hrTime = timestamp ? timeInputToHrTime(timestamp) : now;
+      this.hrTimeObserved = observedTimestamp
+        ? timeInputToHrTime(observedTimestamp)
+        : now;
+    }
+
```

Comment:
How about this?

```
this.hrTime = timestamp ? timeInputToHrTime(timestamp) : hrTime();
this.hrTimeObserved = observedTimestamp
  ? timeInputToHrTime(observedTimestamp)
  : hrTime();
```

Reply:
- `hrTime()` uses `performance.now()` inside which should not be used frequently.
- If it is called twice, `timestamp` and `observedTimestamp` will be different which is not desired.

Reply:
We have actually discussed deprecating/removing the hrTime method entirely. It has real downsides that are very difficult to understand and explain. You can look at #3355, #3134, #3384, #3434, and https://github.com/open-telemetry/opentelemetry-js/issues/3279#issuecomment-1320064182 for some context if you care to find out all the gritty details.